client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx

```diff
@@ -17,26 +18,38 @@ type StepRouteProps = {
 	renderStep: ( step: StepperStep ) => JSX.Element | null;
 };
 
-//TODO: Check we can move RenderStep function to here and remove the renderStep prop
+// TODO: Check we can move RenderStep function to here and remove the renderStep prop
 const StepRoute = ( { step, flow, showWooLogo, renderStep }: StepRouteProps ) => {
 	const userIsLoggedIn = useSelector( isUserLoggedIn );
-	const loginUrl = useLoginUrlForFlow( { flow } );
 	const stepContent = renderStep( step );
-	const shouldRedirectToLogin = step.requiresLoggedInUser && ! userIsLoggedIn;
-	const shouldSkipRender = shouldRedirectToLogin || ! stepContent;
+	const loginUrl = useLoginUrlForFlow( { flow } );
+	const shouldAuthUser = step.requiresLoggedInUser && ! userIsLoggedIn;
+	const shouldSkipRender = shouldAuthUser || ! stepContent;
+	const skipTracking = shouldAuthUser || ! stepContent;
+
+	const useBuiltItInAuth = flow.__experimentalUseBuiltinAuth;
 
 	useStepRouteTracking( {
 		flowName: flow.name,
 		stepSlug: step.slug,
-		skipTracking: shouldSkipRender,
+		skipTracking,
 		flowVariantSlug: flow.variantSlug,
 	} );
 
 	useEffect( () => {
-		if ( shouldRedirectToLogin ) {
+		if ( shouldAuthUser && ! useBuiltItInAuth ) {
 			window.location.assign( loginUrl );
 		}
-	}, [ loginUrl, shouldRedirectToLogin ] );
+	}, [ loginUrl, shouldAuthUser, useBuiltItInAuth ] );
+
+	if ( useBuiltItInAuth && shouldAuthUser && ! userIsLoggedIn ) {
+		return (
+			<div className={ clsx( 'step-route', 'user', flow.name, flow.variantSlug, flow.classnames ) }>
```

Comment:
What do you think about adding a new react-router route to handle the signup and loading using asyncComponet or the react-router lazy?

I think it would be helpful to avoid adding more CSS + JS just to have the signup flow when the user is already logged in.